Philips.CodeAnalysis.SecurityAnalyzers/LicenseAnalyzer.cs

```diff
@@ -11,33 +18,395 @@ namespace Philips.CodeAnalysis.SecurityAnalyzers
 	public class LicenseAnalyzer : SingleDiagnosticAnalyzer
 	{
 		private const string Title = @"Avoid Packages with Unacceptable Licenses";
-		public const string MessageFormat = @"Package '{0}' has an unacceptable license. Consider adding to Allowed.Licenses.txt if license is acceptable.";
-		private const string Description = @"Packages with unacceptable licenses (e.g., copyleft licenses like GPL) should be reviewed before use to ensure compliance with project requirements.";
+		public const string MessageFormat = @"Package '{0}' version '{1}' has an unacceptable license '{2}'. " +
+											 @"Consider adding to Allowed.Licenses.txt if license is acceptable.";
+		private const string Description = @"Packages with unacceptable licenses (e.g., copyleft licenses like GPL) should be " +
+										   @"reviewed before use to ensure compliance with project requirements.";
 
 		public const string AllowedLicensesFileName = @"Allowed.Licenses.txt";
+		public const string LicensesCacheFileName = @"licenses.json";
+
+		// Default acceptable licenses (permissive licenses that are generally safe to use)
+		private static readonly HashSet<string> DefaultAcceptableLicenses = new(StringComparer.OrdinalIgnoreCase)
+		{
+			"MIT",
+			"Apache-2.0",
+			"BSD-2-Clause",
+			"BSD-3-Clause",
+			"ISC",
+			"Unlicense",
+			"0BSD"
+		};
+
+		private static readonly char[] LineSeparators = { '\r', '\n' };
 
 		public LicenseAnalyzer()
 			: base(DiagnosticId.AvoidUnlicensedPackages, Title, MessageFormat, Description, Categories.Security, isEnabled: false)
 		{ }
 
 		protected override void InitializeCompilation(CompilationStartAnalysisContext context)
 		{
-			// Register a dummy syntax node action to satisfy analyzer validation
-			// In a full implementation, this would analyze project files or metadata
-			// to check package license information
-			context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.CompilationUnit);
+			context.RegisterSyntaxNodeAction(AnalyzeProject, SyntaxKind.CompilationUnit);
+		}
+
+		private void AnalyzeProject(SyntaxNodeAnalysisContext context)
+		{
+			// Only analyze if we're not in a test project (to avoid noise during development)
+			if (Helper.ForTests.IsInTestClass(context))
```

Comment:
Because we RegisterSyntaxNodeAction on SyntaxKind.CompilationUnit, will IsInTestClass work? Don't we want a Helper.ForTests method such as "IsTestProject"?